### PR TITLE
New packages: yarn and yarn-bin

### PR DESCRIPTION
**Pull request by kulinacs**
Unfortunately, as yarn requires itself to build, the yarn-bin package is required.

### CODE REVIEW

File: srcpkgs/yarn/template [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Template file for 'yarn'
+pkgname=yarn
+version=0.27.5
+revision=1
+hostmakedepends="yarn-bin"
+depends="nodejs"
+short_desc="Fast, reliable, and secure dependency management"
+maintainer="Nicklaus McClendon <nicklaus@kulinacs.com>"
+license="2-cause-BSD"
+homepage="https://github.com/yarnpkg/${pkgname}"
```
Comment (the-maldridge) [inline srcpkgs/yarn/template]: homepage should not have any variables in it.

File: srcpkgs/yarn/template [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+do_build() {
+	yarn run build-dist
+}
+
+do_install() {
+	vmkdir /usr/share/yarn/
+	vmkdir /usr/bin
+	vcopy dist/lib/ /usr/share/yarn/lib/
+	vcopy dist/bin/ /usr/share/yarn/bin/
```
Comment (the-maldridge) [inline srcpkgs/yarn/template]: is there a reason this can't be vbin'd ?

File: srcpkgs/yarn-bin/template [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+do_extract() {
+	tar xfz ${XBPS_SRCDISTDIR}/${pkgname}-${version}/yarn-v${version}.tar.gz --strip-components=1 -C ${wrksrc}
+}
+
+do_install() {
+	vmkdir /usr/share/yarn/
+	vmkdir /usr/bin
+	vcopy bin /usr/share/yarn/bin
+	vcopy lib /usr/share/yarn/lib
+	ln -rs ${DESTDIR}/usr/share/yarn/bin/yarn ${DESTDIR}/usr/bin/yarn
```
Comment (the-maldridge) [inline srcpkgs/yarn-bin/template]: Is there a reason this can't be vbin'd?

**Comment by the-maldridge** [comment]
I'm also inclined to believe that the homepage for both of these should be https://yarnpkg.com/

File: srcpkgs/yarn/template (hunk 2 above)
Comment (kulinacs) [inline srcpkgs/yarn/template]: Developer recommends installing to /usr/share/yarn/ 
https://github.com/yarnpkg/yarn/blob/master/scripts/build-deb.sh#L42

File: srcpkgs/yarn-bin/template (hunk 3 above)
Comment (kulinacs) [inline srcpkgs/yarn-bin/template]: Developer recommends installing to /usr/share/yarn/ 
https://github.com/yarnpkg/yarn/blob/master/scripts/build-deb.sh#L42

File: srcpkgs/yarn/template (hunk 1 above)
Comment (kulinacs) [inline srcpkgs/yarn/template]: Updated

**Comment by kulinacs** [comment]
Homepage changed. Developer recommends putting everything in /usr/share/yarn, as the executables expect ../lib to contain the libraries for yarn, and these libraries are architecture independent.

**Comment by kulinacs** [comment]
Reworked to install to /usr/bin and /usr/lib/yarn

**Comment by the-maldridge** [comment]
It's a nodejs package so I fully expect this to be unmaintainable, but it looks decent.

**Pull request closed without merging**